### DISCUSSION TITLE
copied-site: Is support for SFTP password authentication planned?

### DISCUSSION THREAD

**Discussion by super7ramp** [hydraulic-software/conveyor#71] [Q&A] [answered]
Is support for uploading site using SFTP password authentication planned?

It looks like only SSH key authentication is supported currently.

**Comment by mikehearn** [comment, marked as answer]
Correct. At the moment it's not planned. Currently we try to ask for all credentials up front before the build starts, because for some kinds of apps it can be quite slow and we want people to be able to kick it off and then go to lunch. So we'd have to either change htat, know at the start of the build that a password is required so we can ask for it, remember it and then use it later.

Is there some reason you can't set up public key auth?

**Reply by super7ramp** [reply]
Thanks for your answer.

> Correct. At the moment it's not planned.

OK.

> Is there some reason you can't set up public key auth?

I'm using a hosting service which only allows a SFTP with password authentication.

**Reply by mikehearn** [reply]
OK. In that case you'll have to use `conveyor make site` and then use `scp` to copy the files up yourself, unfortunately. It's not much harder though: there's nothing special about Conveyor's upload code.

**Reply by super7ramp** [reply]
Yes, that's what I do. I just wanted to confirm the status. Thanks.
